lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4

```diff
@@ -41,7 +41,11 @@ acceptableNode
     ;
 
 connectNode
-    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location (OptionKeyword TransportKeyword value=location)? streamableNode+
+    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location
+                       (OptionKeyword TransportKeyword value=location)?
+                       (OptionKeyword SizeKeyword DecimalLiteral)?
+                       (OptionKeyword ModeKeyword ModeValue)?
+        streamableNode+
     ;
```

Comment:
If mode is "r" that suggests you can't write, but I presume that's not your intention? If the intention is simply to specify whether or not the file is created, maybe just have "connection option create"?